src/Avalonia.Visuals/Media/SolidColorBrush.cs

```diff
@@ -89,5 +90,32 @@ public override IBrush ToImmutable()
         {
             return new ImmutableSolidColorBrush(this);
         }
+
+        /// <inheritdoc/>
+        public bool Equals(ISolidColorBrush other)
+        {
+            return other != null
+                ? Hash(this) == Hash(other)
+                : false;
+        }
+
```

Comment:
I may be missing something here but it is not obvious how an int hash check can robustly compare an item containing a double and a color

Reply:
I don't think we even need to override methods from `Object`

Reply:
@ahopper not sure what you mean, `Hash(ISolidColorBrush)` calls `(brush.Color, brush.Opacity).GetHashCode();` which hashes both members.

@kekekeks what do you mean?

Reply:
> I may be missing something here but it is not obvious how an int hash check can robustly compare an item containing a double and a color

I've just read a lot of documentation on how to do this 100% correctly. I'll update my PR one more time :) 

GetHashCode
- Should be only implemented on immutable types / where the hash cant change
- Should match if 2 objects could be equal - but this is not a guarantee. Call Equals to be sure

Equals
> You should consider implementing the IEquatable<T> interface to support strongly typed tests for equality. Your IEquatable<T>.Equals implementation should return results that are consistent with Equals.

Reply:
C# / .Net equality is really complicated to get right. I'm glad F# does that 90% of the time for you :)